pkg/resource/v1/chartconfig/desired.go

```diff
@@ -2,8 +2,56 @@ package chartconfig
 
 import (
 	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	chartAPIVersion              = "core.giantswarm.io"
+	chartKind                    = "ChartConfig"
+	chartOperatorChart           = "quay.io/giantswarm/chart-operator-chart"
+	chartOperatorChartConfigName = "chart-operator"
+	chartOperatorChannel         = "stable"
+	chartOperatorRelease         = "chart-operator"
 )
 
+// GetDesiredState returns all desired ChartConfigs for managed guest resources.
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	desiredChartConfigs := make([]*v1alpha1.ChartConfig, 0)
+	{
+		chartOperatorChartConfig := newChartOperatorChartConfig(r.baseClusterConfig, r.projectName)
```

Comment:
@tuommaki This is still WIP and I have a question on `baseClusterConfig`. I thought it would be easier to discuss in a PR.

The `baseClusterConfig` struct has all the data I need but it's passed as configuration to the resource. How do the values get populated from the watch?

Do I need to implement `ToClusterGuestConfigFunc` which is in the `certconfig` resource?

Reply:
Is this maybe what you are looking for: https://github.com/giantswarm/cluster-operator/blob/master/service/service.go#L156 ? Rest of the struct I fill in `certconfig` resource: https://github.com/giantswarm/cluster-operator/blob/master/pkg/resource/v1/certconfig/desired.go#L83 using `ClusterGuestConfig`. If that's what you need, then you do need `ToClusterGuestConfigFunc`. That is an extractor function used to abstract provider specific CRs for "generic" resource implementation.

Reply:
@tuommaki Thanks for the explanation. Yes, it's the data in `ClusterGuestConfig` that I need. Cool I'll use `ToClusterGuestConfigFunc` to get that.